### PR TITLE
feat(web): add brand design systems, card thumbnails, and DESIGN.md side-by-side preview

### PR DESCRIPTION
**Pull request by pftom**
## Summary

- Add 7 brand design systems: `arc`, `canva`, `discord`, `duolingo`, `github`, `huggingface`, `openai` (each with its own `DESIGN.md`).
- Show live showcase HTML thumbnails on the Design Systems tab cards (lazy-loaded, cached across re-renders) so the gallery previews the brand at a glance.
- Add a toggleable DESIGN.md side panel in the preview modal, with a syntax-highlighted spec view (`DesignSpecView`) so spec and rendered showcase sit next to each other — mirroring the styles.refero.design layout.
- Make the preview iframe responsive: render the document at a fixed logical viewport width (default `1280px`) and visually `transform: scale(...)` it to fit the actual stage. Opening the side panel no longer squeezes the showcase into a half-broken responsive breakpoint; it just scales down proportionally.
- Add floating collapse / expand handles on the sidebar boundary (`‹` on the iframe right edge when closed, `›` on the sidebar left edge when open) for direct hide/show in addition to the existing header toggle. Falls back to the header button on mobile / stacked layouts.
- New i18n keys (`ds.specToggle`, `ds.specLoading`, `preview.showSidebar`, `preview.hideSidebar`) added to all 12 locales (`de`, `en`, `es-ES`, `fa`, `ja`, `ko`, `pl`, `pt-BR`, `ru`, `tr`, `zh-CN`, `zh-TW`).

## Test plan

- [ ] `pnpm --filter @open-design/web typecheck` (passes locally).
- [ ] Open Design Systems tab — cards show live thumbnails for each brand.
- [ ] Open the preview modal for any DS — DESIGN.md panel renders next to the showcase, scaling preserves layout.
- [ ] Click the `›` handle on the sidebar edge to collapse, click the `‹` handle on the iframe edge to expand.
- [ ] Resize the window — iframe content scales smoothly without breakpoint thrashing.
- [ ] Switch language to each of the 12 locales and confirm the new strings render.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by lefarcen — COMMENTED** [review]
Hey @pftom! Solid feature — the side-by-side DESIGN.md preview mirrors styles.refero.design nicely, and the responsive scaling keeps showcases intact when the sidebar opens. Clean implementation.

## Code review (Lens A)

**Security & correctness:**
- ✅ DesignSpecView renders markdown tokens as CSS-classed React elements (no `dangerouslySetInnerHTML`) — XSS-safe
- ✅ srcDoc iframes go through `buildSrcdoc` sandbox shims
- ✅ Scale logic correctly clamps to [0, 1] and handles zero-width gracefully
- ✅ i18n coverage: all 12 locales received the 4 new keys (`ds.specToggle`, `ds.specLoading`, `preview.showSidebar`, `preview.hideSidebar`)

**Architecture & React patterns:**
- ✅ Lazy-load pattern consistent with ExamplesTab (thumbs cache, IntersectionObserver)
- ✅ `sidebarToggleRef` ref capture prevents effect re-fire on every parent render — good
- ✅ Sidebar toggle callback fires only on state change (not on every render)

**Performance:**
- ✅ IntersectionObserver with 200px rootMargin avoids mounting 120 iframes up front
- ✅ iframe pointer-events:none prevents accidental interaction in thumbnails

**Testing:**
- ⚠️ P2: No test coverage for the new UI paths — `DesignSpecView` markdown parsing, sidebar toggle, scale logic, thumbnail lazy-load. Project has `apps/web/src/artifacts/*.test.ts` but components have no `__tests__`. Consider adding component tests or e2e coverage for the new modal sidebar + gallery card interaction.

## Reasoning review (Lens B — on the 7 brand DESIGN.md files)

**Structure & completeness:**
- ✅ All 7 specs follow consistent schema: Visual Theme, Color Palette, Typography, Components, Spacing, Motion
- ✅ Color palettes are quantified (hex + role names)
- ✅ Typography tables include font stack, size, weight, line-height, letter-spacing
- ✅ Component examples (buttons, cards, inputs) specify padding/radius/shadow/hover states

**Provenance & references:**
- ❓ P3: None of the DESIGN.md files cite official brand guidelines or design system repos (e.g., GitHub Primer docs, Discord's brand assets, Duolingo's style guide). Without citations, it's unclear if these are reverse-engineered from screenshots or based on public docs. Consider adding a "References" section with links to:
  - GitHub → [Primer design system](https://primer.style/foundations/color)
  - Discord → [Blurple brand assets](https://discord.com/branding)
  - Canva → [Brand guidelines](https://www.canva.com/brand/)
  - etc.
  
  This helps future contributors verify token accuracy and distinguishes "official" from "community-derived" specs.

**Assumptions & open questions:**
- ❓ P3: Accessibility (contrast ratios, WCAG conformance) not mentioned in any spec. Brand design systems usually ship with a11y tables (e.g., which color pairs meet AA/AAA). Consider a new section `## Accessibility` with:
  - Which text/bg pairs meet WCAG AA?
  - Focus ring colors and min contrast
  - Motion preferences (prefers-reduced-motion)
  
  Or state explicitly "a11y compliance TBD" if it's out of scope for v1.

**Industry precedent:**
- ✅ Structure matches other public design system docs (Chakra, MUI, Radix)
- ✅ GitHub spec accurately captures Primer's Blurple-free, hairline-border aesthetic

---

**Overall:** Feature is production-ready as-is. The P2 test gap is the only "should-fix" — everything else is P3 polish (citations, a11y notes). Great work on the i18n completeness and XSS-safe markdown rendering.

File: apps/web/src/components/PreviewModal.tsx [hunk 1]
```diff
@@ -115,6 +162,23 @@ export function PreviewModal({
...
+  // mid-breakpoint reflows in the showcase HTML.
+  useEffect(() => {
+    const el = stageFrameRef.current;
+    if (!el) return;
+    const measure = () => {
+      const r = el.getBoundingClientRect();
+      setStageSize({ w: r.width, h: r.height });
+    };
+    measure();
+    const ro = new ResizeObserver(measure);
```
Comment (mrcfps) [inline apps/web/src/components/PreviewModal.tsx]: `ResizeObserver` is now constructed unconditionally when the modal mounts. That breaks environments that do not provide it: the current CI failure is the existing `tests/preview-modal-fullscreen.test.tsx` suite throwing `ReferenceError: ResizeObserver is not defined` at this line before any fullscreen assertions can run. It also means older embedded WebViews would crash the preview modal instead of just missing the scale observer. Please guard this feature, e.g. call `measure()` first and then either use `ResizeObserver` when `typeof ResizeObserver !== 'undefined'`, or fall back to a `window.resize` listener/no-op cleanup in environments without it.

**Review by mrcfps — CHANGES_REQUESTED** [review]
I found one blocking issue that is already breaking the workspace test job. Please add a compatibility path for environments without ResizeObserver before merging.

<sub>Generated by Looper 0.0.0-dev · runner=reviewer · agent=opencode</sub>

**Review by lefarcen — COMMENTED** [review]
Hey @pftom, the ResizeObserver guard looks good — exactly what @mrcfps asked for. 👍

**What's fixed:**
- ✅ Guards `new ResizeObserver(...)` with `typeof ResizeObserver !== 'undefined'`
- ✅ Falls back to `window.resize` listener (jsdom + older WebViews now work)
- ✅ Calls `measure()` first before any observer setup
- ✅ Should fix the CI test failure

**Bonus improvement:**
- Added `contentKey` to sidebar props to re-trigger fetch when swapping design systems while sidebar stays open — nice catch on the UX edge case.

The fix is clean and matches the suggestion from the prior review. No new issues spotted. Ready for maintainer merge when they're satisfied. 🚀

**Review by mrcfps — APPROVED** [review]
